### PR TITLE
methods to normalize ids and keys

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: event-validator.js [hunk 1]
```diff
@@ -66,3 +66,27 @@ module.exports.withOptions = opts => validateEvent.bind(null, opts)
...
 module.exports.hasValidLength = e => !e.body || Buffer.byteLength(JSON.stringify(e.body), 'utf8') <= 32768
+
+// ensure key will have valid sql name
+module.exports.normalizeKey = key => key.replace(/^\$/, '').replace(/[^a-z0-9_]/ig, '_').toLowerCase()
+
+// ensure id will fit in 40 characters
+let crypto
+module.exports.normalizeId = id => {
+
+  if(!crypto) { crypto = require('crypto') }
```
Comment (nathanboktae) [inline event-validator.js]: Actually nevermind, let's just require crypto at the top.

File: event-validator.js [hunk 2]
```diff
@@ -66,3 +66,27 @@ module.exports.withOptions = opts => validateEvent.bind(null, opts)
...
+
+  let result
+  if(id) {
+
+    result = id
+    if(result.length > 40) {
+
```
Comment (nathanboktae) [inline event-validator.js]: can we keep the formatting consistent in this file? Sorry but the hapi style just isn't as popular.

**Comment by nathanboktae** [comment]
👍

File: event-validator.js (hunk 1 above)
Comment (ghost) [inline event-validator.js]: will do.

File: event-validator.js (hunk 2 above)
Comment (ghost) [inline event-validator.js]: no worries!

**Pull request merged**